TypeCobol/Compiler/Preprocessor/ImportedTokensDocument.cs

```diff
@@ -44,17 +47,67 @@ public ImportedTokensDocument(CopyDirective copyDirective, ProcessedTokensDocume
+            bool allowWhitespaceTokens = (forceAllowWhitespaceTokens || HasReplacingDirective && CopyDirective.HasGroupToken);
+            ITokensLinesIterator sourceIterator = ProcessedTokensDocument.GetProcessedTokensIterator(SourceDocument.TextSourceInfo, SourceDocument.Lines, this.CompilerOptions, allowWhitespaceTokens);
             if (HasReplacingDirective
 #if EUROINFO_RULES
                 || (this.CompilerOptions.UseEuroInformationLegacyReplacingSyntax && (this. CopyDirective.RemoveFirst01Level || CopyDirective.InsertSuffixChar))
 #endif
                 )
             {
                 ITokensLinesIterator replaceIterator = new ReplaceTokensLinesIterator(sourceIterator, CopyDirective, CompilerOptions);
-                return replaceIterator;
+                if (CopyDirective.HasGroupToken)
+                {
+                    //Compute the Scanning State
+                    TypeCobol.Compiler.Scanner.MultilineScanState state = null;
+                    TypeCobol.Compiler.Parser.CodeElementsLine cel = SourceDocument.Lines.Count > 0 ? ((TypeCobol.Compiler.Parser.CodeElementsLine)SourceDocument.Lines[0]) : null;
+                    state = cel?.InitialScanState;
+
+                    //Create a Preprocessed text fragment
+                    StringBuilder sb = new StringBuilder();
+                    Token t;
+                    bool bFirst = true;
+                    int line = -1;
+                    ITokensLine tokenLine = null;
+                    while ((t = replaceIterator.NextToken()) != Token.END_OF_FILE)
+                    {
+                        if (bFirst)
+                        {
+                            tokenLine = t.TokensLine;
+                            line = t.Line;
+                        }
+                        if ((tokenLine != t.TokensLine /*|| line != t.Line*/) || bFirst)
+                        {
+                            if (!bFirst)
+                                sb.Append(Environment.NewLine);
+                            sb.Append(new string(' ', Math.Max(0, t.Column - 1)));
+                        }
+                        bFirst = false;
+                        line = t.Line;
+                        tokenLine = t.TokensLine;
+                        sb.Append(t.Text);
+                    }
+                    string preprocessedFRagment = sb.ToString();
+
+                    //Now reparse the preprocessed fragment
+                    ITextDocument initialTextDocumentLines = new ReadOnlyTextDocument(SourceDocument.TextSourceInfo.Name, DocumentFormat.RDZReferenceFormat.Encoding, DocumentFormat.RDZReferenceFormat.ColumnsLayout, preprocessedFRagment);
+                    TypeCobolOptions tcOptions = new TypeCobolOptions();                    
+                    tcOptions.AreForCopyParsing = true;//Doing that any "::" will be treated as two tokens
```

Comment:
`tcOptions` is not used.

Reply:
Yes it must be replaced by the new TypeCobolOptions()